Resolve crashes caused by inconsistent capitalization on Windows (fix issue #2036)

Farmadupe opened the pull request:
**Description** Please explain the changes you made here.
Currently ghdl-ls crashes inside libghdl on windows when hdl-prj.json is used due to capitalization issues (issue #2036). This commit updates ghdl-ls to intercept all sources of filenames (hdl_prj.json and language server RPCs) and normalizes them before they are passed to libghdl.

ghdl-ls now appears to work without crashing on Windows, whether or not hdl_prj.json is used.

I have not yet tested the change on linux, but hopefully it should be unaffected, because all changes are protected by runtime guards that detect the OS.

I am not used to python so my code may be unpythonic.

Paebbels reviewed (changes requested):
Please use `pathlib` instead of old `os.path` to solve this problem.

Farmadupe commented:
Changed lsp.py to use Pathlib  to remove quotes.

Also updated function path_to_uri to replace handwritten code with equivalent stdlib functionality.

Now passes testsuite locally on linux. I cannot run the testsuite locally on windows due to temporary libghdl.dll incompatibility (I don't build libghdl.dll myself).

tgingold commented:
I think the recursion in `normalize_rpc_file_uris` might be too heavy, but that's ok.
So a client doesn't need to normalize the path.  Interesting...
